### DISCUSSION TITLE
Is Python supported for post-processing?

### DISCUSSION THREAD

**Discussion by KrishnaRekapalli** [githubocto/flat-action#5] [Q&A] [answered]
I am wondering if we can use Python for post-processing the downloaded flat files

**Comment by idan** [comment, marked as answer]
Hey there, it is! After a fashion.

Check out @pierrotsmnrd's example repo: https://github.com/pierrotsmnrd/flat_data_py_example

cf. https://twitter.com/githubOCTO/status/1394905517705687045

The short version is:

1. commit your python to the repo
2. add a postprocessing.ts which contains `Deno.run(["python", "mypythonscript.py"])`

**Reply by KrishnaRekapalli** [reply]
Thanks this is helpful! In one of the nonprofit work I was involved, we were using GitHub actions to fetch data and do some post processing on Python and statistical simulation on R. I think Flat actions would be very helpful there.

**Reply by mskyttner** [reply]
Can Deno run a "docker" command? Looking for an example to run R in the processing step, through a docker shell out if possible...

**Reply by irealva** [reply]
I haven't tried this before but an easier route might be to use Deno to execute a bash script that then calls the docker command. [Something like this](https://github.com/aborruso/flat_data_bash_example). 

I know it's a bit of workaround but we haven't implemented the functionality for Flat postprocessing to call on other file types beyond Deno.

**Reply by mskyttner** [reply]
How can I add R to the the execution context for Deno? I guess the steps run on a "ubuntu-latest" container in a GitHub Actions runner? If it is running as a container, can I run another container from within that container? Or would I rather have to add R in a separate step before I could use Deno to shell out to R? I guess python and bash is included in "ubuntu-latest", which allows Deno to run bash and python?
